Source/WebKit/UIProcess/Network/NetworkProcessProxy.cpp

```diff
@@ -592,12 +592,10 @@ void NetworkProcessProxy::terminateWebProcess(WebCore::ProcessIdentifier webProc
         process->requestTermination(ProcessTerminationReason::RequestedByNetworkProcess);
 }
 
-void NetworkProcessProxy::terminateUnresponsiveServiceWorkerProcesses(WebCore::ProcessIdentifier processIdentifier)
+void NetworkProcessProxy::processHasUnresponseServiceWorker(WebCore::ProcessIdentifier processIdentifier)
 {
-    if (auto process = WebProcessProxy::processForIdentifier(processIdentifier)) {
+    if (auto process = WebProcessProxy::processForIdentifier(processIdentifier))
         process->disableRemoteWorkers(RemoteWorkerType::ServiceWorker);
-        process->requestTermination(ProcessTerminationReason::ExceededCPULimit);
```

Comment:
I am trying to make sure I understand the implications of removing this when it runs for a process that happens to be currently suspended (although maybe that would be almost impossible if we do [PR 35932](https://github.com/WebKit/WebKit/pull/35932)):

1. On the UIProcess side we'll clear out the worker info so that WebProcessProxy doesn't think it has a service worker anymore.
2. The WebContent process P won't be woken up to handle the termination event (since e.g. `WebSWContextManagerConnection::Close` is a one-way message), instead it'll handle the termination events once it resumes and gets to handling that message. Also, this process will not get terminated for failing responsiveness checks, since background responsiveness checks are disabled for suspended processes.
3. If another client needs this service worker, then the SW will start up in a different process and process P from (2) will learn about it through some queued IPC message.

Reply:
1 and 2 are correct (although PR 35932 will likely change this).
As of 3, the SW may go to whatever process is deemed appropriate.
If process P wakes up and asks for the service worker, it will likely be selected again.